### PR TITLE
[DRAFT] [NOT TO MERGE] add benchmark tests for traceprocessor

### PR DESCRIPTION
**Pull request by asreehari-splunk**
#### Link to tracking issue
Fixes #

### CODE REVIEW

File: processor/batchprocessor/batch_processor_test.go [hunk 1]
```diff
@@ -80,6 +80,91 @@ func TestProcessorLifecycle(t *testing.T) {
...
 }
 
+func BenchmarkBatchTraceProcessor(b *testing.B) {
+	b.StopTimer()
+	cfg := Config{
+		Timeout:                  100 * time.Millisecond,
+		SendBatchSize:            1000,
+		MetadataCardinalityLimit: defaultMetadataCardinalityLimit,
+	}
+	runTraceProcessorWoMetadataBenchmark(b, cfg)
```
Comment (dmitryax) [inline processor/batchprocessor/batch_processor_test.go:90]: why do we call different functions in the benchmarks? the idea is to have only `MetadataKeys:             []string{"X-Sf-Token"},` line different between benchamarks

File: processor/batchprocessor/batch_processor_test.go [hunk 2]
```diff
@@ -80,6 +80,91 @@ func TestProcessorLifecycle(t *testing.T) {
...
+	require.NoError(b, batcher.Start(context.Background(), componenttest.NewNopHost()))
+
+	b.StartTimer()
+	b.RunParallel(func(pb *testing.PB) {
+		for pb.Next() {
+			requestCount := 100
+			spansPerRequest := 10
+			sentResourceSpans := ptrace.NewTraces().ResourceSpans()
+			for requestNum := 0; requestNum < requestCount; requestNum++ {
+				td := testdata.GenerateTraces(spansPerRequest)
```
Comment (dmitryax) [inline processor/batchprocessor/batch_processor_test.go:111]: Please move `td` building outside. We don't need to benchmark it. We need to benchmark `batcher.ConsumeTraces(context.Background(), td))` only

**Comment by dmitryax** [comment]
This is not intended for merging

File: processor/batchprocessor/batch_processor_test.go (hunk 1 above)
Comment (asreehari-splunk) [reply processor/batchprocessor/batch_processor_test.go:90]: in the second case, I am putting the metadata in the context. i did not want to add a if else, so i separated it

Comment (dmitryax) [reply processor/batchprocessor/batch_processor_test.go:90]: We don't need to put metadata in the context. We are benchmarking if it's safe to add `MetadataKeys: "X-Sf-Token"` by default for non-passthrough (typical) workloads

File: processor/batchprocessor/batch_processor_test.go [hunk 3]
```diff
@@ -80,6 +80,94 @@ func TestProcessorLifecycle(t *testing.T) {
...
+		MetadataCardinalityLimit: defaultMetadataCardinalityLimit,
+	}
+	runTraceProcessorWoMetadataBenchmark(b, cfg)
+}
+
+func runTraceProcessorWoMetadataBenchmark(b *testing.B, cfg Config) {
+	ctx := context.Background()
+
+	sink := new(consumertest.TracesSink)
+	cfg.SendBatchSize = 128
```
Comment (dmitryax) [inline processor/batchprocessor/batch_processor_test.go:97]: Please remove this line

**Pull request closed without merging**